src/main/java/sandbox/SqliteIndexedStorageUtility.java

```diff
@@ -193,8 +192,8 @@ public T getRecordForValue(String fieldName, Object value)
             if (!resultSet.next()) {
                 throw new NoSuchElementException();
             }
-            byte[] mBytes = resultSet.getBytes(org.commcare.modern.database.DatabaseHelper.DATA_COL);
-            return readFromBytes(mBytes);
+            byte[] mBytes = resultSet.getBytes(DatabaseHelper.DATA_COL);
```

Comment:
That's a good catch in general.

For our purposes we treat these records as always being quite small (I think < 1mb) intentionally so I think we can get away without streaming these as-is.